feat: docusaurus migration

alexiscolin opened the pull request:
Replace VuePress docs for Docusaurus per **support to move docs.ignite.com to Docusaurus** [design request](https://www.notion.so/allinbits/support-to-move-docs-ignite-com-to-Docusaurus-8f2919e1b90643bf9af29905bec15524) 
Fabulous preview https://ignite-go-docs--pr2493-feat-docusaurus-migr-nlg7pfz4.web.app/
See opportunities or bugs? Add them to the [Docusaurus Testing before release](https://www.notion.so/allinbits/support-to-move-docs-ignite-com-to-Docusaurus-8f2919e1b90643bf9af29905bec15524#7ba76f16cf1044aca161bec245d5606f) table in the design request.

ilgooz commented on docs/.vuepress/config.js:
```diff
@@ -1,134 +0,0 @@
-      "script",
-      {
-        async: true,
-        src: "https://www.googletagmanager.com/gtag/js?id=G-XL9GNV1KHW",
```
Do we have analytics enabled in the new template?

barriebyron replied:
good question @ilgooz 
I don't know much about analytics in the template... is there action required from my end?

alexiscolin replied:
Not yet. But thank you for the reminder!

alexiscolin replied:
Done 👍

barriebyron commented:
I love the feature of sidebar navigation when the web browser screen is narrow! is there hover text available for this feature/icon? 
<img width="721" alt="image" src="https://user-images.githubusercontent.com/10067215/168666730-617bfd23-b4e9-42ee-8f10-db241a6064c5.png">

barriebyron commented:
Since most of the discussion surrounding this implementation is taking place in the design request on Notion, I just want to say publicly here that @alexiscolin is delivering a much-improved docs site! Our content is part of the product and is essential for developer adoption 
WOW great job design team!! thanks

fadeev commented:
Looks good! Shall we proceed with this PR?

fadeev commented:
As I understand what needs to be done is:

- [ ] Set a base URL for the docs as `/docs`
- [ ] Make `ignite.com` repo watch the `docs` directory in the Ignite CLI repo. When there are changes to the `docs`, clone, build the docs and create a PR
- [ ] Set up a redirect from `docs.ignite.com/*` to `ignite.com/docs/*`

fadeev reviewed (changes requested):
- [ ] Make the logo in the top left (`‹ Ignite`) a link to `ignite.com`

<img width="853" alt="Screen Shot 2022-05-31 at 16 08 33" src="https://user-images.githubusercontent.com/332151/171169831-f188f973-5f46-4564-8ac0-5e2374405c3a.png">

ilgooz commented:
@alexiscolin can we resolve the conflicts please? 🙏

ilgooz reviewed (changes requested):
Reminder to myself (@alexiscolin you can also take a look 🙏), to update:
* https://github.com/ignite-hq/cli/blob/b2ccd8a0af9233ace495be1d8dd4771ae39e88e3/ignite/internal/tools/gen-cli-docs/main.go#L22
* https://github.com/ignite-hq/cli/blob/b2ccd8a0af9233ace495be1d8dd4771ae39e88e3/scripts/gen-cli-docs#L3

barriebyron commented:
@alexiscolin do you need my review on a specific part of this PR? I see conflicts that need to be resolved, but I don't see the conflicts when I check out the branch.
lmk how I can help! 
the docs preview looks amazing

ilgooz commented:
Hey @barriebyron we will soon merge this PR, can you please create another PR for your changes? Thank you.

ilgooz reviewed (commented):
I was comparing old and new config files. I understand that it probably won't be a 1:1 match. 

But still looks like some links are missing for ex. I cannot find the Youtube channel link/icon in the new website as well as some other links like [these](https://github.com/ignite-hq/cli/blob/e7dc69305b75c805f6df50f046c432195f3d2a81/docs/.vuepress/config.js#L101-L114) and maybe more.

Old [config](https://github.com/ignite-hq/cli/blob/e7dc69305b75c805f6df50f046c432195f3d2a81/docs/.vuepress/config.js#L101-L114)

New [config](https://github.com/ignite-hq/cli/blob/d4ced532cd8fd995a336bc3aa810536f6474a2c1/docs/docusaurus.config.js).

---

Another Q, why these index files are dupilicated?
* https://github.com/ignite-hq/cli/blob/d4ced532cd8fd995a336bc3aa810536f6474a2c1/docs/src/pages/_index.md

* https://github.com/ignite-hq/cli/blob/d4ced532cd8fd995a336bc3aa810536f6474a2c1/docs/docs/index.md

Other these, LGTM.

barriebyron commented:
> 

@ilgooz no specific changes from me, I was attempting to resolve conflicts that the PR introduced

alexiscolin commented:
@ilgooz, thanks for your feedbacks.
Concerning the config/icon matching, it mainly because the design has changed and some of the icons are not present in the [Figma file](https://www.figma.com/file/EnSPW5Prj2fzvl1z6Shlds/Ignite-%7C-Docusaurus-%7C-UI?node-id=3176%3A204650) I used as model. So, I started "from scratch" to follow this Figma layout, playing with Docusaurus constraints that are not he same as VuePress ones. But If you think we need more buttons or features, I would be happy to extend the config.

Your second question is about the duplicated file. I've to admit that is a legacy file I have forgotten... At first, I created the entry point as a Docusaurus Page but after some tests, I decided to move forward with a Doc index... Thank you for reporting it, I'm going to remove it.

alexiscolin commented:
@barriebyron All good for me. I'm only waiting for the Algolia snippet but maybe we can ship it anyway? (@fadeev, @ilgooz )

fadeev commented:
We can ship Algolia later but as a temporary solution enabling local search (in a separate PR) could be great.

ilgooz commented:
Actually, if there is no a fundamental limitation (e.g. Algolia has to index us with the actual domain name or temporary local search is not possible to do in this PR), I'm in favor of not introducing the new website yet.

I believe we should have the search functionality in this PR otherwise, we'll be just introducing a -way- better UI but also removing an existing feature for no a real reason, which is not ideal.

alexiscolin commented:
I've added all links that were present in the current docs website (ressources links from the sidebar and YouTube link from the footer). I guess the others links from the VuePress config file are outdated legacy code, since they seem not used in the [cosmos template](https://github.com/cosmos/vuepress-theme-cosmos). 

I've also added a local search engine as a temporary replacement for Algolia.

fadeev reviewed (approved):
Excellent work!

barriebyron commented:
@alexiscolin 
Looks so good! I think the topic titles can be improved and changed so we don't have redundant words

I strongly recommend a breadcrumb separator other than the slash (/) character 
how about a > so we see  `Docs > Knowledge Base`
instead of `Docs / Knowledge Base`
<img width="323" alt="image" src="https://user-images.githubusercontent.com/10067215/172933116-6528159d-10f7-43ba-bce4-6651d169f9ae.png">

barriebyron commented:
@alexiscolin 
did you see https://allinbits.slack.com/archives/C014GFM3FRQ/p1654782205161579 that we finally got our Algolia docSearch
